### PR TITLE
[SYCL][Graph] Support for kernel-bundle

### PR DESCRIPTION
**Pull request by mfrancepillois**
- Adds support for using kernel-bundle in Graph.
- Adds tests checking that kernel bundles can be used with the Record&Replay API and the Explicit API.

### CODE REVIEW

File: sycl/source/handler.cpp [hunk 1]
```diff
@@ -816,7 +816,11 @@ void handler::verifyUsedKernelBundle(const std::string &KernelName) {
     return;
 
   kernel_id KernelID = detail::get_kernel_id_impl(KernelName);
-  device Dev = detail::getDeviceFromHandler(*this);
+  device Dev;
```
Comment (aelovikov-intel) [inline sycl/source/handler.cpp:819]: I think this default ctor isn't a no-op. Can we combine variable declaration and initialization in single statement? Either ternary operator or an "Immediately Invoked Function Expression".

File: sycl/source/detail/scheduler/commands.cpp [hunk 2]
```diff
@@ -2388,8 +2388,31 @@ pi_int32 enqueueImpCommandBufferKernel(
   pi_program PiProgram = nullptr;
```
Comment (aelovikov-intel) [inline sycl/source/detail/scheduler/commands.cpp:2388]: @sergey-semenov , would you be able to look at the changes in this file? I'm not familiar enough to provide comprehensive review.

File: sycl/source/handler.cpp (hunk 1 above)
Comment (mfrancepillois) [reply sycl/source/handler.cpp:819]: We updated it and now use a ternary operator.

**Review by reble — APPROVED** [review]
LGTM

File: sycl/source/detail/scheduler/commands.cpp (hunk 2 above)
Comment (EwanC) [reply sycl/source/detail/scheduler/commands.cpp:2388]: @aelovikov-intel @sergey-semenov Could one of you approve this PR if you are happy with the content, thanks.

**Review by aelovikov-intel — APPROVED** [review]

**Pull request merged**